File: ContactsManagerAppWithJavaROOMDBMVVM/app/src/main/java/com/masterclass/contactsmanagerappwithjavaroomdbmvvm/activities/AddNewContactActivity.java
package com.masterclass.contactsmanagerappwithjavaroomdbmvvm.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.R;
import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.activityClickHandlers.AddContactsClickHandler;
import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.database.Contacts;
import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.databinding.ActivityAddNewContactBinding;
import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.viewModels.MyViewModel;

public class AddNewContactActivity extends AppCompatActivity {

    ActivityAddNewContactBinding binding;
    AddContactsClickHandler clickHandler;
    Contacts contact;

    MyViewModel myViewModel;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_new_contact);

        contact = new Contacts();
        binding = DataBindingUtil.setContentView(this, R.layout.activity_add_new_contact);

        myViewModel = new ViewModelProvider(this).get(MyViewModel.class);

        clickHandler = new AddContactsClickHandler(contact, this, myViewModel);
        binding.setContact(contact);
        binding.setClickHandler(clickHandler);

    }
}

File: ContactsManagerAppWithKotlinROOMDBMVVM2/app/src/main/java/com/masterclass/contactsmanagerappwithkotlinroomdbmvvm/activities/AddNewContactActivity.kt
package com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.R
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.activityClickHandlers.AddContactsClickHandler
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.database.Contacts
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.databinding.ActivityAddNewContactBinding
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.viewModels.MyViewModel

class AddNewContactActivity : AppCompatActivity() {

    private lateinit var binding: ActivityAddNewContactBinding
    private lateinit var clickHandler: AddContactsClickHandler
    private lateinit var contact: Contacts
    private lateinit var myViewModel: MyViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_add_new_contact)

        contact = Contacts()


        myViewModel = ViewModelProvider(this)[MyViewModel::class.java]

        clickHandler = AddContactsClickHandler(contact, this, myViewModel)
        binding.contact = contact
        binding.clickHandler = clickHandler
    }
}
